options.go

```diff
@@ -54,6 +55,13 @@ func WithNamespaces(namespaces ...string) ServerOption {
 	})
 }
 
+// WithUpstreamOptions registers Temporal server options.
+func WithUpstreamOptions(options ...temporal.ServerOption) ServerOption {
+	return newApplyFuncContainer(func(cfg *liteconfig.Config) {
+		cfg.UpstreamOptions = options
```

Comment:
I _think_ later options can override previous ones, so appending here might make it easier to compose options together.

```suggestion
		cfg.UpstreamOptions = append(cfg.UpstreamOptions, options...)
```